src/utils.ts

```diff
@@ -289,12 +289,14 @@ export async function getSourceUnits(rootPath: string, contractsDirectories: str
   return sourceUnits;
 }
 
-export function smockableNode(node: ASTNode): boolean {
+export function smockableNode(node: ASTNode, contract: ContractDefinition): boolean {
   if (node instanceof VariableDeclaration) {
     // If the state variable is constant then we don't need to mock it
     if (node.constant || node.mutability === 'immutable') return false;
     // If the state variable is private we don't mock it
     if (node.visibility === 'private') return false;
+  } else if (node instanceof FunctionDefinition) {
+    if (node.isConstructor && contract.abstract) return false;
```

Comment:
Can we access the contract as `parent` of the node?

Reply:
Yeah its a good improvement